### PR TITLE
feat: setting to toggle insomnia user-agent [INS-4107]

### PR DESCRIPTION
**Pull request by filfreire**
This PR adds a new setting that allows disabling adding the default insomnia app version user-agent to new requests.

image

related to INS-4107

### CODE REVIEW

File: packages/insomnia/src/main/network/curl.ts [hunk 1]
```diff
@@ -136,6 +136,7 @@ const openCurlConnection = async (
     const start = performance.now();
     const clientCertificates = await models.clientCertificate.findByParentId(options.workspaceId);
     const filteredClientCertificates = filterClientCertificates(clientCertificates, options.url, 'https:');
+    console.log('[filipe]', options.suppressUserAgent);
```
Comment (filfreire) [inline packages/insomnia/src/main/network/curl.ts]:
```suggestion

```

File: packages/insomnia/src/main/network/libcurl-promise.ts [hunk 2]
```diff
@@ -384,6 +384,7 @@ export const createConfiguredCurlInstance = ({
 
   const userAgent: RequestHeader | null = headers.find((h: any) => h.name.toLowerCase() === 'user-agent') || null;
   const userAgentOrFallback = typeof userAgent?.value === 'string' ? userAgent?.value : 'insomnia/' + version;
+  console.log('[filipe] User-Agent:', userAgentOrFallback, req.suppressUserAgent);
```
Comment (filfreire) [inline packages/insomnia/src/main/network/libcurl-promise.ts]:
```suggestion
```

**Review by gatzjames — COMMENTED** [review]
https://github.com/Kong/insomnia/pull/7640/files#diff-0d876a97684cc6c868835d083785c52894f879ce84216e92bdb0a9876d4abcafR577
I think the setting should be passed like the other objects here instead of using `models.settings.getOrCreate()` inside this function

**Comment by filfreire** [comment]
@gatzjames, I submitted a second pass, please review again

**Review by gatzjames — DISMISSED** [review]
Nice one!
Tested with many variations of enabling/disabling this and works great!

File: packages/insomnia/src/main/network/libcurl-promise.ts [hunk 3]
```diff
@@ -382,12 +382,22 @@ export const createConfiguredCurlInstance = ({
...
 
-  const userAgent: RequestHeader | null = headers.find((h: any) => h.name.toLowerCase() === 'user-agent') || null;
-  const userAgentOrFallback = typeof userAgent?.value === 'string' ? userAgent?.value : 'insomnia/' + version;
-  curl.setOpt(Curl.option.USERAGENT, userAgentOrFallback);
-  if (req.suppressUserAgent) {
-    curl.setOpt(Curl.option.USERAGENT, '');
+  // cleanup orphan app/version user-agent headers
+  const existingAppVersionUserAgent: RequestHeader | null = headers.find((h: any) => h.name.toLowerCase() === 'user-agent' && h.disabled === false && h.value.startsWith('insomnia')) || null;
+  if (existingAppVersionUserAgent) {
+    headers.splice(headers.indexOf(existingAppVersionUserAgent), 1);
```
Comment (jackkav) [inline packages/insomnia/src/main/network/libcurl-promise.ts]: splice is a mutating function, and tough to reason about. 
Its better to use a pure function and name the output to make your code easier to read.

**Comment by subnetmarco** [comment]
Let's update this in such a way that it does not add the default Insomnia `User-Agent` to new requests that I create (but, if a different user that I am collaborating with is adding the `User-Agent`, then this checkbox will not do anything since it was not created by me).

The checkbox should be renamed with "Do not add Insomnia User-Agent for new requests".

**Review by jackkav — DISMISSED** [review]

**Review by jackkav — APPROVED** [review]

**Pull request merged**